packages/respect-core/src/modules/flow-runner/prepare-request.ts

```diff
@@ -153,15 +156,30 @@ export async function prepareRequest(
     collectSecretFields(ctx, schema, groupParametersValuesByName(parameters, param.in), [name]);
   }
 
-  const evaluatedBody = evaluateRuntimeExpressionPayload({
+  let evaluatedBody = evaluateRuntimeExpressionPayload({
     payload: requestBody,
     context: expressionContext,
     contentType,
   });
 
+  if (replacements) {
+    // To handle string replacement properly with variables it's better to parse
+    // the string into an object and process the replacement.
+    // Also resolves query string variables before sending.
+    if (typeof evaluatedBody === 'string') {
+      evaluatedBody = querystring.parse(evaluatedBody);
```

Comment:
I see it was like that before but I'm confused why we parse with queryString here?